Dev

Snay110 opened the pull request:
Implemented doctor list UI with Supabase integration instead of mock server, added full auth with modals, and updated doctor detail page UI

Snay110 committed:
feat:Added UI to DoctorDetail

Snay110 committed:
init:первый комит

Snay110 committed:
feat(auth): add login/register modals and supabase client setup

Snay110 committed:
feat: UI home and Modal authorization windows in Heder

Snay110 committed:
feat: integrate Supabase and update modal UI

Snay110 committed:
init: last commit/auth

Snay110 committed:
feat: refactor doctor list display and tweak doctor detail refactor

Maxfri commented on code/my-app/src/app/pages/doctor-details/details.pages.tsx:
```diff
@@ -1,29 +1,80 @@
+  const [error, setError] = useState<string | null>(null);
+
   const { id } = useParams<{ id: string }>();
-  const { data, isLoading, isError, error } = useGetDoctorByIQuery(id!, {
```
Почему решил убрать запрос через tanstack query?

Maxfri commented on code/my-app/src/app/pages/doctor-details/details.pages.tsx:
```diff
@@ -1,29 +1,80 @@
-  });
+  const navigate = useNavigate();
+
+  useEffect(() => {
```
Тогда не надо было бы добавлять юзэффект

Maxfri commented on code/my-app/src/app/pages/doctor/doctor.pages.tsx:
```diff
@@ -1,31 +1,54 @@
+import type { Doctor } from "@/features/auth/api/doctorListApi";
 
 export default function DoctorPage() {
-  const { data, isLoading, isError, error } = useGetDoctorsQuery();
```
тут также) У тебя три состояния useState вместо одного запроса)

Maxfri commented on code/my-app/src/features/header/AppHeader.pages.tsx:
```diff
@@ -1,21 +1,45 @@
-          href={ROUTES.DOCTORS}
-        />
+        <Link
+          className=" bg-gray-800  hover:bg-gray-500 text-white px-4 py-2 rounded-md shadow-md  "
```
пустые строки у className вначале и конце

Maxfri commented on code/my-app/src/features/header/AppHeader.pages.tsx:
```diff
@@ -1,21 +1,45 @@
+          Doctors
+        </Link>
+        <button
+          className="  bg-gray-800  hover:bg-gray-500 text-white px-4 py-2 rounded-md shadow-md "
```
и тут

Maxfri commented on code/my-app/src/features/header/AppHeader.pages.tsx:
```diff
@@ -1,21 +1,45 @@
+          login
+        </button>
+        <button
+          className="  bg-gray-800  hover:bg-gray-500 text-white px-4 py-2 rounded-md shadow-md "
```
и тут

Snay110 committed:
refactor:post-PR editing and updating of the doctor's personal page interface

Snay110 commented:
Replaced mock server data with Supabase (tables + auth), added login/signup modals, refreshed UI for doctor list and doctor profile, and switched from RTK Query to SWR with caching.
